lib/Doctrine/Migrations/Tools/Console/Command/DoctrineCommand.php

```diff
@@ -68,35 +69,22 @@ protected function outputHeader(
 
     protected function initialize(InputInterface $input, OutputInterface $output) : void
     {
-        if ($this->dependencyFactory !== null) {
-            $this->dependencyFactory->freeze();
-
-            return;
-        }
+        if ($this->dependencyFactory === null) {
+            $config        = is_string($input->getOption('configuration')) ? $input->getOption('configuration'): null;
```

Comment:
In which case(s) is it neither a string nor `null`? Shouldn't we error out in that kind of case?

Reply:
it is to make phpstan happy. it detects that `configuration` can  be an integer

Reply:
I would have expected `phpstan-symfony` to take care of that, especially since we did the extra bit of setup required for options to be properly typed:

https://github.com/doctrine/migrations/blob/ab7ebf3fecb796d2c48641ae08dd97d3553f696e/phpstan.neon.dist#L26-L27

Reply:
it does, but not that well. it detects is as `int|string|null` instead of `string|null`.

Reply:
Ok apparently this will not help distinguish strings and integers:

https://github.com/phpstan/phpstan-symfony/blob/f793b68b4665f223620769e8c3ea4dbf40c53ce4/tests/Type/Symfony/InputInterfaceGetOptionDynamicReturnTypeExtensionTest.php#L29-L39

That makes sense, because I don't know how it would achieve that…